src/include/cache.i

```diff
@@ -339,9 +339,9 @@ __wt_eviction_needed(
 	 * we involve the application thread.
 	 */
 	if (pct_fullp != NULL)
-		*pct_fullp = (u_int)WT_MAX(0, 100 - WT_MIN(
-		    (int)cache->eviction_trigger - (int)pct_full,
-		    (int)cache->eviction_dirty_trigger - (int)pct_dirty));
+		*pct_fullp = (double)WT_MAX(0.0, 100.0 - WT_MIN(
```

Comment:
"(double)WT_MAX" that casting seems like redundant, since both arguments to WT_MAX are double. The arguments to WT_MIN, involve subtraction where one operand is double, so the other should implicitly cast to double, and you shouldn't need an explicit casting.